tests/cases/focus/keyboard_focus.slint

```diff
@@ -0,0 +1,159 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+// cSpell: ignore ogoffart tronical
+
+import { SpinBox, LineEdit, HorizontalBox, VerticalBox, TabWidget } from "std-widgets.slint";
+
+TestCase := Window {
+    preferred-width: 500px;
+    preferred-height: 400px;
+
+    property<[{name: string, account: string, score: float}]> model: [
+        {
+            name: "Olivier",
+            account: "ogoffart",
+            score: 456,
+        },
+        {
+            name: "Simon",
+            account: "tronical",
+            score: 789,
+        }
+    ];
+
+    property<string> result;
+
+    VerticalBox {
+        SpinBox {
+            key-pressed(event) => {
+                if (event.text == "X") {
+                    result += "T:";
+                    return accept;
+                }
+                return reject;
+            }
+            value: 23;
+        }
+
+        for person[i] in model: HorizontalBox {
+            SpinBox {
+                key-pressed(event) => {
```

Comment:
Aw.....   We should never have exposed that in the public API.

I'd rather you use a FocusScope directly.

Reply:
And the second downside of `SpinBox` being a `FocusScope` directly is that when "re-implementing" key-pressed, there's probably no way of calling the original implementation.

We've had this a few times, when a widget in the fluent style wasn't a Rectangle but something else that offered functionality of some kind, like the `GroupBox` being a vertical layout, or `SpinBox` or `ComboBox` being a `FocusScope`.

Maybe it's worth a comment here at least, and when we change `SpinBox` the compiler will hopefully point us to this code needing changing.